### PR TITLE
fix: Change string.Format style to string interpolation

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
- Changed string.Format to string interpolation format. string.Format is bit expensive operatino.
- Added thread safe support for ForcedVariation.

## Test plan
- FPS shouldn't break it.

### CODE REVIEW

File: OptimizelySDK/Bucketing/DecisionService.cs [hunk 1]
```diff
@@ -46,7 +46,12 @@ public class DecisionService
         /// set by the user by calling setForcedVariation (it is not the same as the    
         /// whitelisting forcedVariations data structure in the Experiments class). 
         /// </summary>
-        private Dictionary<string, Dictionary<string, string>> ForcedVariationMap;
+#if NET35
+private Dictionary<string, Dictionary<string, string>> ForcedVariationMap;
+#else
+private System.Collections.Concurrent.ConcurrentDictionary<string, Dictionary<string, string>> ForcedVariationMap;
```
Comment (msohailhussain) [inline OptimizelySDK/Bucketing/DecisionService.cs:52]: ThreadSafe dictionary. 

TODO: Add unit test

**Comment by msohailhussain** [comment]
Sure, I will share results.

**Comment by msohailhussain** [comment]
@mikeng13 I tested it separately and don't see any major difference. For more than 150K+ requests it took 0.5 seconds for this log. 
string.Format(@"{0} Variation ""{1}"" is mapped to experiment ""{2}"" and user ""{3}"" in the forced variation map", Logger.DEBUG, experimentKey, variationKey, i.ToString());
and I see more or less the same time (avg 0.445 sec)
$@"{Logger.DEBUG} Variation ""{ experimentKey}"" is mapped to experiment ""{variationKey}"" and user ""{i.ToString()}"" in the forced variation map";

**Comment by mikeng13** [comment]
Thanks

**Pull request merged**